src/utils/intrusive_ptr_test.cc

```diff
@@ -80,6 +85,13 @@ TEST(IntrusivePtrTest, CopyConstructorIncreasesCountAndSharesPtr) {
   RefCountedTypePtr copy = ptr;
   EXPECT_EQ(value->count(), 2);
   EXPECT_EQ(ptr.get(), copy.get());
+
+  auto derived_ptr = make_intrusive_ptr<RefCountedTypeDerived>();
+  EXPECT_EQ(derived_ptr->count(), 1);
+  intrusive_ptr<RefCountedType> base_ptr(derived_ptr);
+  EXPECT_EQ(derived_ptr.get(), base_ptr.get());
+  EXPECT_EQ(derived_ptr->count(), 2);
+  EXPECT_EQ(base_ptr->count(), 2);
```

Comment:
Move this to a separate test say `TEST(IntrusivePtrTest, CanConvertBetweenConvertiblePtrs)` or something like that.  

Also check that the underlying pointers are the same. `EXPECT_EQ(base_ptr, derived_ptr);`?